packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -459,41 +460,55 @@ const ActionMenuItem = ({
   selected,
   type,
   value,
+  onClick,
+  onKeyDown,
   leadingVisual: LeadingVisual,
   ...props
 }: ActionMenuItemAnchorProps | ActionMenuItemDefaultProps) => {
   const {size} = useActionMenuContext()
 
-  const isAnchor = as === 'a'
-  const InnerTag = isAnchor ? 'a' : React.Fragment
+  const handleClick = useCallback(
+    (e: React.MouseEvent<HTMLLIElement>) => {
+      onClick?.(e)
 
-  return (
-    <li
-      className={clsx(
+      if (!disabled) {
+        handler?.(String(value))
+      }
+    },
+    [handler, value, disabled, onClick],
+  )
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLLIElement>) => {
+      onKeyDown?.(e)
+
+      if (!disabled && e.key === 'Enter') {
+        handler?.(String(value))
+      }
+    },
+    [handler, value, disabled, onKeyDown],
+  )
+
+  const liProps = useMemo<React.HTMLProps<HTMLLIElement> & {'data-value': unknown}>(
+    () => ({
+      className: clsx(
         styles.ActionMenu__item,
         type === 'single' && styles[`ActionMenu__item--selection-type-${type}`],
         size && styles[`ActionMenu__item--${size}`],
         className,
-      )}
-      role={roleTypeMap[type || 'single']}
-      aria-checked={as === 'a' || type === 'none' ? undefined : selected ? 'true' : 'false'}
-      aria-disabled={disabled ? 'true' : 'false'}
-      onClick={!isAnchor && handler && !disabled ? () => handler(String(value)) : undefined}
-      onKeyDown={
-        !isAnchor && handler && !disabled ? event => event.key === 'Enter' && handler(String(value)) : undefined
-      }
-      tabIndex={0}
-      data-value={value}
-      {...(isAnchor ? {} : props)}
-    >
-      <InnerTag
-        {...(isAnchor
-          ? {
-              className: clsx(styles['ActionMenu__item-anchor'], disabled && styles['ActionMenu__item--disabled']),
-              href: props.href,
-            }
-          : {})}
-      >
+      ),
+      role: roleTypeMap[type || 'single'],
+      'aria-disabled': disabled,
+      tabIndex: -1,
```

Comment:
Changed to `-1` to prevent items from being focusable via <kbd>Tab</kbd> as focus is controlled using arrow keys